### PR TITLE
[Perf] opt nsa backend init forward metada

### PR DESCRIPTION
**Pull request by Baidu-AIAK**
## Motivation
During Dsa backend init forward metadate, it use for-loop to get the  Seqlen_expanded. when BS is large, It's very time-consuming.
image

## Modifications
We replaced the for-loop implementation with a vectorized implementation.

## Accuracy Tests
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/attention/nsa_backend.py [hunk 1]
```diff
@@ -399,25 +399,10 @@ def init_forward_metadata(self, forward_batch: ForwardBatch):
             extend_seq_lens_cpu = [self.speculative_num_draft_tokens] * batch_size
             forward_batch.extend_seq_lens_cpu = extend_seq_lens_cpu
 
-            seqlens_int32_cpu = [
-                self.speculative_num_draft_tokens + kv_len
-                for kv_len in forward_batch.seq_lens_cpu.tolist()
-            ]
-            seqlens_expanded = torch.cat(
```
Comment (Fridge003) [inline python/sglang/srt/layers/attention/nsa_backend.py:411]: Is it possible to replace these loops with a single triton kernel, which can be even better

Comment (Baidu-AIAK) [reply python/sglang/srt/layers/attention/nsa_backend.py:411]: now it is triton kernel.

**Comment by Fridge003** [comment]
@Baidu-AIAK Please fix lint with
```
pre-commit install
pre-commit run --all-files
```

**Comment by Fridge003** [comment]
@Baidu-AIAK Can you add accuracy result for GPQA and AIME25: https://docs.sglang.io/basic_usage/deepseek_v32.html#accuracy-test-with-gpqa-diamond

Also post some data on mtp accept length, to show this PR doesn't cause regression on MTP case.

**Comment by Fridge003** [comment]
Merged in #19536
